### ISSUE TITLE
[package] openal/1.19.1: Cannot cross-compile for Windows on Linux

### ISSUE THREAD

**Issue by rsulli55** [conan-io/conan-center-index#1222]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **openal/1.19.1**
  * Operating System+version: **Linux Manjaro 19.0.2 (fully updated)**
  * Compiler+version: **x86_64-w64-mingw32-gcc 9.3.0**
  * Conan version: **conan 1.23.0**
  * Python version: **Python 3.8.2**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
Configuration for profile win64:

[settings]
os_build=Linux
arch_build=x86_64
os=Windows
arch=x86_64
compiler=gcc
compiler.version=9.3
compiler.libcxx=libstdc++11
build_type=Release
[options]
[build_requires]
[env]
CONAN_CMAKE_FIND_ROOT_PATH=/usr/bin/x86_64-w64-mingw32
CONAN_CMAKE_TOOLCHAIN_FILE=~/Development/C++/DiceRoller/cmake/x86_64-w64-mingw32.cmake
CHOST=x86_64-w64-mingw32
AR=x86_64-w64-mingw32-ar
AS=x86_64-w64-mingw32-as
RANLIB=x86_64-w64-mingw32-ranlib
CC=x86_64-w64-mingw32-gcc
CXX=x86_64-w64-mingw32-g++
STRIP=x86_64-w64-mingw32-strip
RC=x86_64-w64-mingw32-windres
```
### Steps to reproduce (Include if Applicable)
1. Save the above profile as `win64`
2. Insert
```
[requires]
openal/1.19.1
```
into `conanfile.txt`
3. Run `conan install ./ --profile win64 --build missing`

**NOTE**: It is possible this will fail due to a CMake issue, but that is **not** the problem I am having (please see **real issue** below).

The CMake issue is that CMake passes the flag -rdynamic to every GNU compiler on Linux, but `w64-mingw32` does not recognize this flag.  If the compilation fails with 
```
The C compiler
    /usr/bin/x86_64-w64-mingw32-gcc
  is not able to compile a simple test program.
```
Edit 
`/usr/share/cmake-3.17/Modules/Platform/Linux-GNU.cmake` (or your analogous CMake file)
by removing `set(CMAKE_SHARED_LIBRARY_LINK__FLAGS -rdynamic)`
and adding 
```
if(MINGW)
   set(CMAKE_SHARED_LIBRARY_LINK__FLAGS` )
  else()
  set(CMAKE_SHARED_LIBRARY_LINK__FLAGS -rdynamic)
endif()
```

### Real Issue
When cross-compiling openal, it expects to build the native-tools target with the build systems compiler.

_Source_: See [here](https://github.com/kcat/openal-soft/issues/150)  where kcat mentions:

> it essentially executes a second cmake process without the cross-compiler, so it builds native executables which can run to generate some files

However, with the `win64` profile from above conan uses the `w64-mingw32` compiler for the native-tools target.  Is there a way to change the environment variables in the `conanfile.py` to make it so that the native-tools target will use the build systems toolchain?  

### Logs (Include/Attach if Applicable)
Relevant make output (you can see it is using **/usr/bin/x86_64-w64-mingw32-gcc** for native-tools target)
<details><summary>Click to expand log</summary>

```
[  1%] Generating native-tools/bin2h, native-tools/bsincgen
[  2%] Building C object source_subfolder/CMakeFiles/common.dir/common/almalloc.c.obj
[  4%] Building C object source_subfolder/CMakeFiles/common.dir/common/alcomplex.c.obj
[  5%] Building C object source_subfolder/CMakeFiles/common.dir/common/atomic.c.obj
[  7%] Building C object source_subfolder/CMakeFiles/common.dir/common/rwlock.c.obj
-- The C compiler identification is GNU 9.3.0
[  8%] Building C object source_subfolder/CMakeFiles/common.dir/common/threads.c.obj
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/x86_64-w64-mingw32-gcc
-- Check for working C compiler: /usr/bin/x86_64-w64-mingw32-gcc - works
-- Detecting C compiler ABI info
[ 10%] Building C object source_subfolder/CMakeFiles/common.dir/common/uintmap.c.obj
-- Detecting C compiler ABI info - failed
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/x86_64-w64-mingw32-g++
-- Check for working CXX compiler: /usr/bin/x86_64-w64-mingw32-g++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pow in m
-- Looking for pow in m - found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools
make[3]: Entering directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
make[4]: Entering directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
make[5]: Entering directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
make[5]: Entering directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
Scanning dependencies of target bin2h
Scanning dependencies of target bsincgen
make[5]: Leaving directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
make[5]: Leaving directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
make[5]: Entering directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
make[5]: Entering directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
[ 25%] Building C object CMakeFiles/bin2h.dir/bin2h.c.o
[ 11%] Linking C static library ../lib/libcommon.a
[ 50%] Building C object CMakeFiles/bsincgen.dir/bsincgen.c.o
[ 75%] Linking C executable bin2h
make[2]: Leaving directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder'
make[5]: Leaving directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
[ 11%] Built target common
[ 75%] Built target bin2h
[100%] Linking C executable bsincgen
make[5]: Leaving directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
[100%] Built target bsincgen
make[4]: Leaving directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
make[3]: Leaving directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools'
make[2]: Leaving directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder'
[ 11%] Built target native-tools
make[2]: Entering directory '/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder'
[ 15%] Generating default-44100.mhr.h
[ 15%] Generating bsinc_inc.h
[ 15%] Generating default-48000.mhr.h
/bin/sh: native-tools/bsincgen: No such file or directory

```
In the directory `/home/ryan/.conan/data/openal/1.19.1/_/_/build/344886eda55829e935447d0708e3b993938b32c8/build_subfolder/source_subfolder/native-tools`

an `ls` shows that `bin2h` and `bsincgen` are there but are Windows executables
``` $ ls
bin2h.exe  bsincgen.exe  CMakeCache.txt  CMakeFiles  cmake_install.cmake  Makefile
```
</details>

Please let me know if more information is needed.  Thank you!

**Comment by famouzkk** [comment]
Bump, same issue here.

**Comment by JStarFighter** [comment]
Same issue. Can we specifiy something in the profile for this to done correctly?

**Comment by danimtb** [comment]
Hi! I was able to build the package in WIndows with MinGW, so that is a first step that it is compiling there. I will try crossbuilding from linux as well and report back. Thanks!

**Comment by danimtb** [comment]
Testing with other versions of the library this seems to be working fine cross-building from linux to windows MinGW.
- Versions 1.22.2, 1.21.1, 1.21.0 work fine cross-building.
- Version 1.20.1 fails with same error of `-rdynamic`.

The version 1.29.1 is old, it was released 2018 according to https://www.openal-soft.org/openal-releases/ and there are much more recent versions. I think it is not worth solving this issue as the maintainers of the library already solved the issue since 1.21.0 version.

**Comment by danimtb** [comment]
The release notes of 1.21.1 say that this limitation was fixed:
```
Got rid of the compile-time native-tools targets, using cmake and global
    initialization instead. This should make cross-compiling less troublesome.
```
So that confirms that the issue was in the library itself and not in the recipe.

Could you try using a newer version for that? Is this still an issue for you @rsulli55 @JStarFighter @famouzkk ?
Thank you
